app/utils/JourneyChecker.scala

```diff
@@ -32,23 +33,45 @@ trait JourneyChecker extends IncomeSourcesUtils {
   implicit val ec: ExecutionContext
 
   val sessionService: SessionService
-  def withIncomeSourcesFSWithSessionCheck(journeyType: JourneyType)(codeBlock: => Future[Result])(implicit user: MtdItUser[_], hc: HeaderCarrier): Future[Result] = {
+
+  def withCustomSession(journeyType: JourneyType)(codeBlock: => Future[Result])
+                       (implicit user: MtdItUser[_], hc: HeaderCarrier): Future[Result] = {
     withIncomeSourcesFS {
-      journeyChecker(journeyType).flatMap {
-        case true => user.userType match {
-          case Some(Agent) => Future.successful(Redirect(controllers.incomeSources.add.routes.YouCannotGoBackErrorController.showAgent(journeyType.businessType)))
-          case _ => Future.successful(Redirect(controllers.incomeSources.add.routes.YouCannotGoBackErrorController.show(journeyType.businessType)))
+      customSessionStarted(journeyType).flatMap { state =>
```

Comment:
Can we not get the whole data UIJourneySessionData and pass back to the code block, so we only have to do this once? Otherwise we are duplicating the mongodb calls...
something like:
```
withSessionData {
   sessionData => 
      ...stuff
}
```